src/attr/_make.py

```diff
@@ -450,6 +450,9 @@ def _transform_attrs(
     if field_transformer is not None:
         attrs = field_transformer(cls, attrs)
 
+        if inspect.isgenerator(attrs):
```

Comment:
Instead of generator specifically, would it be better to make this cover all iterators instead? A simple check would be `if hasattr(attrs, "__next__")`

Reply:
Perhaps,  I always feel icky looking at dunders like that though.

WDYT @hynek ?

Reply:
Does anything speak against calling tuple unconditionally? The inspect isn't free either.

Reply:
That if you returned something else odd (like a single value? I dunno) you 'd end up with that wrapped in a tuple.

Either way it's going to behave oddly or give you an error. 

Happy to just make it a tuple unconditionally. Perhaps worht an `if not isinstance(attrs, (list,tuple))` check too?